### PR TITLE
added test file test_positions.py

### LINKED ISSUE #74: Create test cases for position endpoints
1. Read guideline [here](https://gist.github.com/djeck1432/f722f5fe518c80b248ff609627da25e5)
2. Set up the project, read [README.md](https://github.com/djeck1432/spotnet/blob/main/README.md#development-environment-setup) how to do that.
3. Add test cases for next positive and negative scenarios for next endpoints [here](https://github.com/djeck1432/spotnet/blob/main/web_app/api/position.py): 
- `open_position`
- `close_position`
- `get_repay_data`
- `create_position_with_transaction_data`

### PR DESCRIPTION
**Pull request by binayak9932**
fixes  #74

### CODE REVIEW

File: web_app/tests/test_positions.py [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+    Returns:
+        None
+    """
+    position_id = "valid_position_id"
+    with patch(
+        "web_app.db.crud.PositionDBConnector.open_position"
+    ) as mock_open_position:
+        mock_open_position.return_value = "Position successfully opened"
+        response = client.get(f"/api/open-position?position_id={position_id}")
+        assert response.status_code == 200
```
Comment (djeck1432) [inline web_app/tests/test_positions.py:32]: Please, use shorter option. response.ok

File: web_app/tests/test_positions.py [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+from fastapi.testclient import TestClient
+from httpx import AsyncClient
+from web_app.api.main import app
+from web_app.contract_tools.mixins.deposit import DepositMixin
+
+client = TestClient(app)
+app.dependency_overrides.clear()
+
+
+# Test cases for /api/open-position
```
Comment (djeck1432) [inline web_app/tests/test_positions.py:14]: Remove this comment, 
fyi: after inserts code from ChatGPT, it would be good if you remove some unnecessary comments

File: web_app/tests/test_positions.py [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+    Returns:
+        None
+    """
+    position_id = "valid_position_id"
+    with patch(
+        "web_app.db.crud.PositionDBConnector.close_position"
+    ) as mock_close_position:
+        mock_close_position.return_value = "Position successfully closed"
+        response = client.get(f"/api/close-position?position_id={position_id}")
+        assert response.status_code == 200
```
Comment (djeck1432) [inline web_app/tests/test_positions.py:70]: Use `response.ok` please

File: web_app/tests/test_positions.py [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+    """
+    Test for successfully closing a position using a valid position ID.
+
+    Args:
+        client (AsyncClient): The test client for the FastAPI application.
+
+    Returns:
+        None
+    """
+    position_id = "valid_position_id"
```
Comment (djeck1432) [inline web_app/tests/test_positions.py:64]: valid position id it's `uuid` str representation

File: web_app/tests/test_positions.py [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+
+    Args:
+        client (AsyncClient): The test client for the FastAPI application.
+
+    Returns:
+        None
+    """
+    supply_token = "valid_supply_token"
+    wallet_id = "valid_wallet_id"
+    mock_repay_data = {
```
Comment (djeck1432) [inline web_app/tests/test_positions.py:111]: Please, use: `pytest.mark.parametrize` and add more than only one example of data. This data is very important, we want to make sure, that we covered as much as possible cases

File: web_app/tests/test_positions.py [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+    """
+    Test for successfully creating a position with valid form data.
+
+    Args:
+        client (AsyncClient): The test client for the FastAPI application.
+
+    Returns:
+        None
+    """
+    form_data = {
```
Comment (djeck1432) [inline web_app/tests/test_positions.py:201]: Please, use: `pytest.mark.parametrize` and add more than only one example of data. This data is very important, we want to make sure, that we covered as much as possible cases

File: web_app/tests/test_positions.py [hunk 7]
```diff
@@ -0,0 +1,282 @@
...
+        ) as mock_get_transaction_data,
+        patch(
+            "web_app.db.crud.PositionDBConnector.get_contract_address_by_wallet_id"
+        ) as mock_get_contract_address,
+    ):
+        mock_create_position.return_value = mock_position
+        mock_get_transaction_data.return_value = mock_deposit_data
+        mock_get_contract_address.return_value = "mock_contract_address"
+        response = client.post("/api/create-position", json=form_data)
+        assert response.status_code == 200
```
Comment (djeck1432) [inline web_app/tests/test_positions.py:243]: Use, `response.ok`

File: web_app/tests/test_positions.py [hunk 8]
```diff
@@ -0,0 +1,282 @@
...
+    """
+    Test for attempting to create a position with invalid input data, which should return a 422 error.
+
+    Args:
+        client (AsyncClient): The test client for the FastAPI application.
+
+    Returns:
+        None
+    """
+    response = client.post(
```
Comment (djeck1432) [inline web_app/tests/test_positions.py:277]: add more edge cases, token_symbol is incorrect or amount is `str` type not `int`, add test cases, where `mulitpiler` is `str` instead of `int`, or None
You should try insert for `int` type params -> random `str` or None

**Review by djeck1432 — CHANGES_REQUESTED** [review]

**Comment by binayak9932** [comment]
@djeck1432 can you please review

File: web_app/tests/test_positions.py [hunk 9]
```diff
@@ -0,0 +1,431 @@
...
+    """
+    Test for successfully creating a position with valid form data.
+
+    Args:
+        client (AsyncClient): The test client for the FastAPI application.
+
+    Returns:
+        None
+    """
+    form_data = {
```
Comment (djeck1432) [inline web_app/tests/test_positions.py:338]: You Maybe forgot to remove it

**Review by djeck1432 — CHANGES_REQUESTED** [review]
Please, sync with main branch your code. 
Left one comment, I think you just forgot to do that, and ready to merge it

File: web_app/tests/test_positions.py [hunk 10]
```diff
@@ -0,0 +1,446 @@
...
+    Returns:
+        None
+    """
+    position_id = str(uuid.uuid4())
+    with patch(
+        "web_app.db.crud.PositionDBConnector.close_position"
+    ) as mock_close_position:
+        mock_close_position.return_value = "Position successfully closed"
+        response = client.get(f"/api/close-position?position_id={position_id}")
+        assert response.ok
```
Comment (djeck1432) [inline web_app/tests/test_positions.py:79]: Please, change it to `response.is_success` ,this object, doesn't have `ok` method, sorry

File: web_app/tests/test_positions.py [hunk 11]
```diff
@@ -0,0 +1,446 @@
...
+        )
+        mock_get_contract_address.assert_called_once_with(wallet_id)
+        mock_get_position_id.assert_called_once_with(wallet_id)
+        mock_get_repay_data.assert_called_once_with(supply_token)
+        expected_response = {
+            **mock_repay_data,
+            "contract_address": "mock_contract_address",
+            "position_id": "123",
+        }
+        assert response.ok
```
Comment (djeck1432) [inline web_app/tests/test_positions.py:207]: Please, change it to `response.is_success` ,this object, doesn't have `ok` method, sorry

File: web_app/tests/test_positions.py [hunk 12]
```diff
@@ -0,0 +1,446 @@
...
+        response = await client.post(
+        "/api/create-position-with-transaction-data",
+        data={
+        "wallet_id": wallet_id,
+        "token_symbol": token_symbol,
+        "amount": amount,
+        "multiplier": multiplier,
+            },
+            )   
+        assert response.ok
```
Comment (djeck1432) [inline web_app/tests/test_positions.py:395]: Please, change it to `response.is_success` ,this object, doesn't have `ok` method, sorry

**Review by djeck1432 — APPROVED** [review]
Looks good to me,
